### PR TITLE
Add iDRAC role to control Dell baremetal osp compute nodes

### PR DESCRIPTION
**Pull request by dpinhas**
This role will be used for DELL systems in RH internal network, To allow cifmw to manipulate bios/bmc to configure it for OSP network

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

File: ci_framework/roles/idrac_configuration/tasks/cleanup.yml
Comment (cjeanner) [inline ci_framework/roles/idrac_configuration/tasks/cleanup.yml:1]: this file may be removed.

File: ci_framework/roles/idrac_configuration/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        group: idrac_target_hosts
+      with_items: "{{ host_list }}"
+
+- hosts: idrac_target_hosts
+  connection: local
+  vars_files:
+    - idrac_hosts_vars.yml
+  roles:
+    - idrac_configuration
+
```
Comment (cjeanner) [inline ci_framework/roles/idrac_configuration/README.md:51]: you may remove some of the empty lines

**Review by cjeanner — COMMENTED** [review]
You may want to rebase against `main` - though not mandatory at this stage.

File: ci_framework/roles/idrac_configuration/README.md (hunk 1 above)
Comment (dpinhas) [reply ci_framework/roles/idrac_configuration/README.md]: removed, thanks.

File: ci_framework/roles/idrac_configuration/tasks/cleanup.yml
Comment (dpinhas) [reply ci_framework/roles/idrac_configuration/tasks/cleanup.yml:1]: removed, thanks.

**Comment by cjeanner** [comment]
recheck

**Review by psathyan — APPROVED** [review]
/lgtm

**Pull request merged**